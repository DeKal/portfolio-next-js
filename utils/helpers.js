import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
