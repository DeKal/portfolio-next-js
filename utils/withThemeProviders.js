import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
