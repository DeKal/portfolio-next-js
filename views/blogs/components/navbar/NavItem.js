import styled from 'styled-components/macro'
import Nav from 'react-bootstrap/Nav'

export default styled(Nav.Link)`
  font-size: 18px;
`
