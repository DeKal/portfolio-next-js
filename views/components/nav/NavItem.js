import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

const NavItem = ({ page, isSelected, setSelectedPage }) => {
  return isSelected ? (
    <SelectedContainer>{page}</SelectedContainer>
  ) : (
    <Container
      data-test-id="nav-link"
      href={`#${page}`}
      onClick={() => {
        setSelectedPage(page)
      }}
    >
      {page}
    </Container>
  )
}

NavItem.propTypes = {
  page: PropTypes.string,
  isSelected: PropTypes.bool,
  setSelectedPage: PropTypes.func
}

export default NavItem

const Container = styled.a`
  line-height: ${props => props.theme.navContentMinHeight};
  font-size: ${props => props.theme.fontSizes.XXL};
  transition: color 0.5s;

  &:hover {
    color: ${props => props.theme.colors.colorDarken};
    border: none;
  }

  @media screen and (max-width: 800px) {
    font-size: ${props => props.theme.fontSizes.XL};
    line-height: 2;
    border-bottom: none;
  }
`
const SelectedContainer = styled(Container)`
  color: ${props => props.theme.colors.colorDarken};
  border: none;
`
