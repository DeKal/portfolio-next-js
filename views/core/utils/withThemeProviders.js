import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
