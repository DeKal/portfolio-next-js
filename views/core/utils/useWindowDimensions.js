import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
