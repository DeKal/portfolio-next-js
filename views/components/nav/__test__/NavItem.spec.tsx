import * as React from 'react'
import { shallowWithTheme, mountWithTheme } from '~/utils/withThemeProviders'
import NavItem from '~/views/components/nav/NavItem'
const Router = require('next/router').default;

const actionWithPromise = () => {
  return new Promise((resolve) => resolve());
};

const mockedRouter = {
  push: actionWithPromise,
  replace: actionWithPromise,
  prefetch: () => {},
};

// @ts-ignore
Router.router = mockedRouter

describe('Nav Item', () => {
  it('should render Selected NavItem', function() {
    const setSelectedPage = jest.fn()

    const tree = shallowWithTheme(
      <NavItem page={'Home'} isSelected setSelectedPage={setSelectedPage} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('should render Normal NavItem', function() {
    const setSelectedPage = jest.fn()

    const tree = shallowWithTheme(
      <NavItem page={'Home'} setSelectedPage={setSelectedPage} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('should set selected page when clicking NavItem', function() {
    const setSelectedPage = jest.fn()

    const tree = mountWithTheme(
      <NavItem page={'Home'} setSelectedPage={setSelectedPage} />
    )
    tree.find('a[data-test-id="nav-link"]').simulate('click')
    expect(setSelectedPage.mock.calls.length).toEqual(1)
    expect(setSelectedPage).toHaveBeenCalledWith('Home');
  })
})
