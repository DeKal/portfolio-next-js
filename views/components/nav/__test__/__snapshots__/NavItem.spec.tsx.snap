// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Item should render Normal NavItem 1`] = `
<NavItem__Container
  data-test-id="nav-link"
  href="#Home"
  onClick={[Function]}
>
  Home
</NavItem__Container>
`;

exports[`Nav Item should render Selected NavItem 1`] = `
<NavItem__SelectedContainer>
  Home
</NavItem__SelectedContainer>
`;
